router/core/transport.go

```diff
@@ -85,67 +86,80 @@ func (ct *CustomTransport) RoundTrip(req *http.Request) (*http.Response, error)
-	preHandlers    []TransportPreHandler
-	postHandlers   []TransportPostHandler
-	retryOptions   retrytransport.RetryOptions
-	requestTimeout time.Duration
-	logger         *zap.Logger
+	preHandlers                   []TransportPreHandler
+	postHandlers                  []TransportPostHandler
+	retryOptions                  retrytransport.RetryOptions
+	requestTimeout                time.Duration
+	localhostFallbackInsideDocker bool
+	logger                        *zap.Logger
 }
 
 var _ ApiTransportFactory = TransportFactory{}
 
 type TransportOptions struct {
-	preHandlers    []TransportPreHandler
-	postHandlers   []TransportPostHandler
-	retryOptions   retrytransport.RetryOptions
-	requestTimeout time.Duration
-	logger         *zap.Logger
+	PreHandlers                   []TransportPreHandler
+	PostHandlers                  []TransportPostHandler
+	RetryOptions                  retrytransport.RetryOptions
+	RequestTimeout                time.Duration
+	LocalhostFallbackInsideDocker bool
+	Logger                        *zap.Logger
 }
 
 func NewTransport(opts *TransportOptions) *TransportFactory {
+	if opts.LocalhostFallbackInsideDocker && docker.Inside() {
+		if opts.Logger != nil {
+			opts.Logger.Info("localhost fallback enabled, connections that fail to connect to localhost will be retried using host.docker.internal")
+		}
+	}
+
 	return &TransportFactory{
-		preHandlers:    opts.preHandlers,
-		postHandlers:   opts.postHandlers,
-		logger:         opts.logger,
-		retryOptions:   opts.retryOptions,
-		requestTimeout: opts.requestTimeout,
+		preHandlers:                   opts.PreHandlers,
+		postHandlers:                  opts.PostHandlers,
+		retryOptions:                  opts.RetryOptions,
+		requestTimeout:                opts.RequestTimeout,
+		localhostFallbackInsideDocker: opts.LocalhostFallbackInsideDocker,
+		logger:                        opts.Logger,
 	}
 }
 
 func (t TransportFactory) RoundTripper(transport http.RoundTripper, enableStreamingMode bool) http.RoundTripper {
-	tp := NewCustomTransport(
-		t.logger,
-		trace.NewTransport(
-			transport,
-			[]otelhttp.Option{
-				otelhttp.WithSpanNameFormatter(SpanNameFormatter),
-				otelhttp.WithSpanOptions(otrace.WithAttributes(otel.EngineTransportAttribute)),
-			},
-			trace.WithPreHandler(func(r *http.Request) {
-				span := otrace.SpanFromContext(r.Context())
-				reqContext := getRequestContext(r.Context())
-				operation := reqContext.operation
-
-				if operation != nil {
-					if operation.name != "" {
-						span.SetAttributes(otel.WgOperationName.String(operation.name))
-					}
-					if operation.opType != "" {
-						span.SetAttributes(otel.WgOperationType.String(operation.opType))
-					}
-					if operation.hash != 0 {
-						span.SetAttributes(otel.WgOperationHash.String(strconv.FormatUint(operation.hash, 10)))
-					}
+	if t.localhostFallbackInsideDocker && docker.Inside() {
+		transport = docker.NewLocalhostFallbackRoundTripper(transport)
```

Comment:
Please mention here or in the package description that this module / transport exist only to simplify local development. It should not be used for production. Please also add some docs on gitbook about this smooth integration.

Reply:
Added